skimage/future/graph/rag.py

```diff
@@ -491,31 +499,27 @@ def draw_rag(labels, rag, img, border_color=None, node_color='#ffff00',
     for (n, data), region in zip(rag.nodes_iter(data=True), regions):
         data['centroid'] = region['centroid']
 
+    cc = colors.ColorConverter()
     if border_color is not None:
-        border_color = cc.to_rgb(border_color)
+        border_color = cc.to_rgba(border_color)
         out = segmentation.mark_boundaries(out, rag_labels, color=border_color)
 
-    if colormap is not None:
-        edge_weight_list = [d['weight'] for x, y, d in
-                            rag.edges_iter(data=True) if d['weight'] < thresh]
-        norm = colors.Normalize()
-        norm.autoscale(edge_weight_list)
-        smap = cm.ScalarMappable(norm, colormap)
+    ax.imshow(out)
 
-    for n1, n2, data in rag.edges_iter(data=True):
-
-        if data['weight'] >= thresh:
-            continue
-        r1, c1 = map(int, rag.node[n1]['centroid'])
-        r2, c2 = map(int, rag.node[n2]['centroid'])
-        line = draw.line(r1, c1, r2, c2)
+    edge_weight_list = [d['weight'] for x, y, d in
+                        rag.edges_iter(data=True) if d['weight'] < thresh]
 
-        if colormap is not None:
-            out[line] = smap.to_rgba([data['weight']])[0][:-1]
-        else:
-            out[line] = edge_color
+    norm = colors.Normalize()
+    norm.autoscale(edge_weight_list)
+    smap = cm.ScalarMappable(norm, edge_cmap)
 
-        circle = draw.circle(r1, c1, 2)
-        out[circle] = node_color
+    for n1, n2, data in rag.edges_iter(data=True):
 
-    return out
+        if data['weight'] < thresh:
+            r1, c1 = map(int, rag.node[n1]['centroid'])
+            r2, c2 = map(int, rag.node[n2]['centroid'])
+            col = smap.to_rgba([data['weight']])[0][:-1]
+            ax.add_artist(plt.Line2D([c1, c2], [r1, r2],  color=col,
```

Comment:
I would use a `LineCollection` for this, you get better prefromance and mpl will take care of the norm/color mapping for you.